### ISSUE TITLE
AoT Failing to find app.module.ngfactory after 0.0.47

### ISSUE THREAD

**Issue by zbarbuto** [driftyco/ionic-app-scripts#704]
#### Short description of the problem:

After upgrading past 0.0.46 (0.0.47+) if I build my app with the --aot or --prod flags the build 'succeeds' with no errors or warnings but when running on a device/simulator there is a white screen with the console message "cannot find module './app.module.ngfactory'". 

It seems to be related to the switch to the virtual file system for building because there were no issues using 0.0.46 and if I downgrade the project back to 0.0.46 the AoT works again. I've double checked that we completed all the steps required in the breaking changes of the changelog for 0.0.47.

I've ran the build with debug mode and the ngfactory definitely gets generated (even logged the contents of the file and it's definitely all there) but for whatever reason when the webpack stage runs it fails to find the file in the virtual file system.

Some googling revealed this problem is often related to an incompatible typescript version but I've checked and the project is definitely using 2.0.9.

**Which @ionic/app-scripts version are you using?**

0.0.47, 1.0.0

**Comment by danbucholtz** [comment]
Can you post your `main` file content?

Thanks,
Dan

**Comment by zbarbuto** [comment]
It's as per the docs:

```
import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';

import { AppModule } from './app.module';

platformBrowserDynamic().bootstrapModule(AppModule);
```

**Comment by danbucholtz** [comment]
Can you run `npm run ionic:build --prod --debug` and paste the output here? Seems strange to me. Hmmm.

Thanks,
Dan

**Comment by zbarbuto** [comment]
[output.txt](https://github.com/driftyco/ionic-app-scripts/files/753347/output.txt)

**Comment by danbucholtz** [comment]
```
import { platformBrowser } from '@angular/platform-browser';
import { AppModuleNgFactory } from './app.module.ngfactory';
import { enableProdMode } from '@angular/core';
enableProdMode();
platformBrowser().bootstrapModuleFactory(AppModuleNgFactory);
```

Is the new `main.ts` generated at runtime. That looks correct to me.

Can you check that `node_modules/typescript/package.json` has a `version` of `2.0.x` and not `2.1.x`?

Otherwise, can you upload a repository somewhere so we can recreate the issue?

Thanks,
Dan

**Comment by zbarbuto** [comment]
As mentioned in the OP I've definitely checked that we're using ts 2.0.9.

Unfortunately, since this is a close-source client project I can't post the source code, so I guess we'll just keep using 0.0.46 since that works.

**Comment by danbucholtz** [comment]
Can you recreate the issue in a separate project?

Thanks,
Dan

**Comment by zbarbuto** [comment]
Yep, will give it a go in the next couple of days.

**Comment by fareesh** [comment]
@zbarbuto I had this issue but I had `"typescript": "^2.0.9"`. Removing the ^ character and running `npm install` appears to have solved this for me.

**Comment by danbucholtz** [comment]
It appears this issue is fixed since it's been awhile, so I'm going to close it.

Thanks,
Dan

**Comment by zbarbuto** [comment]
Sorry about that, Dan.

The issue still exists but I haven't got a chance to try to setup a public repo to reproduce this. I'm happy for this to remain closed until I do so and I will @ you for a re-open when I do so. At this stage we're happy to continue using the old version of app-scripts that works for us for now.

Cheers.

**Comment by zbarbuto** [comment]
@danbucholtz So I finally resolved this after many headaches and ripping pieces out of our app until I had it essentially back to the base ionic app... or anyone who finds this issue:

Turns out it was a breaking change in 2.1+ which meant that custom webpack configs don't work the way they did in <= 2.0.9.

I had a custom webpack config as follows:

```javascript
const DefinePlugin = require('webpack/lib/DefinePlugin');

module.exports = {
  plugins: [
    new DefinePlugin({
      'process.env': {
        'NODE_ENV' : JSON.stringify(process.env.NODE_ENV || 'development')
      }
    })
  ]
};
```

In 2.0.9 this would work in conjunction with the usual prod build but in 2.1+ it seems to overwrite the default ionic webpack config completely resulting in my weird issue. I copied the full ionic webpack config from `node_modules/@ionic/app-scripts/config` to my `./config` folder and added the DefinePlugin to that - and 🎉  aot working again in the latest ionic-app-scripts.
